pandas/tests/test_base.py

```diff
@@ -269,52 +269,50 @@ def setup_method(self, method):
-
+    def test_none_comparison(self, series_fixture):
         # bug brought up by #1079
         # changed from TypeError in 0.17.0
-        for o in self.is_valid_objs:
-            if isinstance(o, Series):
-
-                o[0] = np.nan
-
-                # noinspection PyComparisonWithNone
-                result = o == None  # noqa
-                assert not result.iat[0]
-                assert not result.iat[1]
-
-                # noinspection PyComparisonWithNone
-                result = o != None  # noqa
-                assert result.iat[0]
-                assert result.iat[1]
-
-                result = None == o  # noqa
-                assert not result.iat[0]
-                assert not result.iat[1]
-
-                result = None != o  # noqa
-                assert result.iat[0]
-                assert result.iat[1]
-
-                if (is_datetime64_dtype(o) or is_datetimetz(o)):
-                    # Following DatetimeIndex (and Timestamp) convention,
-                    # inequality comparisons with Series[datetime64] raise
-                    with pytest.raises(TypeError):
-                        None > o
-                    with pytest.raises(TypeError):
-                        o > None
-                else:
-                    result = None > o
-                    assert not result.iat[0]
-                    assert not result.iat[1]
+        series_fixture[0] = np.nan
+
+        # noinspection PyComparisonWithNone
+        result = series_fixture == None  # noqa
+        assert not result.iat[0]
+        assert not result.iat[1]
+
+        # noinspection PyComparisonWithNone
+        result = series_fixture != None  # noqa
+        assert result.iat[0]
+        assert result.iat[1]
+
+        result = None == series_fixture  # noqa
+        assert not result.iat[0]
+        assert not result.iat[1]
+
+        # this fails for numpy < 1.9
+        # and oddly for *some* platforms
+        result = None != series_fixture  # noqa
+        assert result.iat[0]
+        assert result.iat[1]
+        if (is_datetime64_dtype(series_fixture) or
+                is_datetimetz(series_fixture)):
+            # Following DatetimeIndex (and Timestamp) convention,
+            # inequality comparisons with Series[datetime64] raise
+            with pytest.raises(TypeError):
+                None > series_fixture
+            with pytest.raises(TypeError):
+                series_fixture > None
+        else:
+            result = None > series_fixture
+            assert not result.iat[0]
+            assert not result.iat[1]
 
-                    result = o < None
-                    assert not result.iat[0]
-                    assert not result.iat[1]
+            result = series_fixture < None
+            assert not result.iat[0]
+            assert not result.iat[1]
 
-    def test_ndarray_compat_properties(self):
+    def test_ndarray_compat_properties(self, indices_fixture, series_fixture):
```

Comment:
so this would cause an enormous number of tests as you get a cartesian product, instead

defined *another* fixture in ``pandas/conftest`` called ``all_indices_or_series_fixture`` which is a concatenation of the series & indices

Reply:
@jreback could you give me some pointers, I tried to concatenate the two fixtures but the result is not as expected

Reply:
instead create a new fixture that uses the input to both and then pass a single fixture
```
indices_list = [tm.makeFloatIndex()......]
series_list = [.....]


@pytest.mark.parametrize(params=indices_list):
def indices_fixture(...):
    ...

# same for series

@pytest.mark.parmetrize(parmas=indices_list + series_list)
def series_and_indices(....)

```

Reply:
I think we need to zip them instead of concatenating, let me try that. Thanks for the tip